lib/modules/manager/maven/index.spec.ts

```diff
@@ -316,12 +317,33 @@ describe('modules/manager/maven/index', () => {
       packages.forEach(({ deps }) => {
         deps.forEach(({ registryUrls }) => {
           const depUrls = new Set([...registryUrls!]);
-          expect(depUrls).toEqual(urls);
+          expect(depUrls).toStrictEqual(urls);
         });
       });
       expect(packages).toMatchSnapshot();
     });
 
+    it('should include registryUrls in the correct order', async () => {
+      fs.readLocalFile
+        .mockResolvedValueOnce(pomContent)
+        .mockResolvedValueOnce(profileSettingsContent);
+      const packages = await extractAllPackageFiles({}, [
+        'simple.pom.xml',
+        'profile.settings.xml',
+      ]);
+      const urls = new Set([
+        'https://repo.adobe.com/nexus/content/groups/public',
+        'https://maven.atlassian.com/content/repositories/atlassian-public/',
+        'https://repo.maven.apache.org/maven2',
+      ]);
+      packages.forEach(({ deps }) => {
+        deps.forEach(({ registryUrls }) => {
+          const depUrls = new Set([...registryUrls!]);
+          expect([...depUrls]).toStrictEqual([...urls]);
+        });
+      });
```

Comment:
```suggestion
      const urls: string[] = [
        'https://repo.adobe.com/nexus/content/groups/public',
        'https://maven.atlassian.com/content/repositories/atlassian-public/',
        'https://repo.maven.apache.org/maven2',
      ];
      packages.forEach(({ deps }) => {
        deps.forEach(({ registryUrls }) => {
          expect(registryUrls).toStrictEqual(urls);
        });
      });
```
Why do you use Sets?